src/gei/Commands/MigrateOrgCommand.cs

```diff
@@ -0,0 +1,143 @@
+
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_SOURCE_PAT env variable or --github-source-pat option. Will fall back to GH_PAT or --github-target-pat if not set."
+            };
+            var githubTargetOrg = new Option<string>("--github-target-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_PAT env variable or --github-target-pat option."
+            };
+            var githubTargetEnterprise = new Option<string>("--github-target-enterprise")
+            {
+                IsRequired = true,
+                Description = "Name of the target enterprise."
+            };
+            var ssh = new Option("--ssh")
+            {
+                IsRequired = false,
+                IsHidden = true,
+                Description = "Uses SSH protocol instead of HTTPS to push a Git repository into the target repository on GitHub."
+            };
+            var githubSourcePat = new Option<string>("--github-source-pat")
+            {
+                IsRequired = false
+            };
+            var githubTargetPat = new Option<string>("--github-target-pat")
+            {
+                IsRequired = false
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(githubSourceOrg);
+            AddOption(githubTargetOrg);
+            AddOption(githubTargetEnterprise);
+
+            AddOption(ssh);
+            AddOption(githubSourcePat);
+            AddOption(githubTargetPat);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<MigrateOrgCommandArgs>(Invoke);
+        }
+
+        public async Task Invoke(MigrateOrgCommandArgs args)
+        {
+            if (args is null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            _log.Verbose = args.Verbose;
+
+            LogAndValidateOptions(args);
+
+            var githubApi = _targetGithubApiFactory.Create(targetPersonalAccessToken: args.GithubTargetPat);
+
+            var githubEnterpriseId = await githubApi.GetEnterpriseId(args.GithubTargetEnterprise);
+            var sourceOrgUrl = GetGithubOrgUrl(args.GithubSourceOrg, null);
+            var sourceToken = GetSourceToken(args);
+            var targetToken = args.GithubTargetPat ?? _environmentVariableProvider.TargetGithubPersonalAccessToken();
+
+            var migrationId = await githubApi.StartOrganizationMigration(
+                sourceOrgUrl,
+                args.GithubTargetOrg,
+                githubEnterpriseId,
+                sourceToken,
+                targetToken);
+
+            _log.LogSuccess($"Org Migration has been initiated (ID: {migrationId}).");
```

Comment:
Are we not going to have the CLI poll for status and give the user some indication of success/failure and/or progress?

Reply:
We need to implement some kind of --wait option for org migrations. As well as a making sure the watch migration command works when give the migration ID. 

Customers will want to monitor the status.

Reply:
I suggest waiting should be the default behavior of migrate-org. Unlike migrate-repo customers won't be generating a migration script when migrating whole orgs (at least I don't think so).

There could be some optional flag to not wait (e.g. --no-wait)

Reply:
I have added the `--wait` option and it's behaviour is similar to the `--wait` option in `migrate-repo` command.